chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/InviteDevelopersJob.java

```diff
@@ -0,0 +1,343 @@
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import gov.healthit.chpl.FeatureList;
+import gov.healthit.chpl.auth.permission.GrantedPermission;
+import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
+import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.dao.UserDeveloperMapDAO;
+import gov.healthit.chpl.dao.auth.InvitationDAO;
+import gov.healthit.chpl.dao.auth.UserPermissionDAO;
+import gov.healthit.chpl.dao.impl.BaseDAOImpl;
+import gov.healthit.chpl.domain.auth.Authority;
+import gov.healthit.chpl.domain.auth.UserInvitation;
+import gov.healthit.chpl.domain.auth.UserPermission;
+import gov.healthit.chpl.domain.concept.CertificationEditionConcept;
+import gov.healthit.chpl.dto.DeveloperDTO;
+import gov.healthit.chpl.dto.UserDeveloperMapDTO;
+import gov.healthit.chpl.email.ChplHtmlEmailBuilder;
+import gov.healthit.chpl.email.EmailOverrider;
+import gov.healthit.chpl.entity.CertificationStatusType;
+import gov.healthit.chpl.entity.developer.DeveloperEntity;
+import gov.healthit.chpl.exception.UserCreationException;
+import gov.healthit.chpl.exception.UserPermissionRetrievalException;
+import gov.healthit.chpl.exception.UserRetrievalException;
+import gov.healthit.chpl.util.Util;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "inviteDevelopersJobLogger")
+public class InviteDevelopersJob implements Job {
+
+    @Autowired
+    private BulkDeveloperEmailer bulkEmailer;
+
+    @Autowired
+    private DeveloperListingMapDao developerListingMapDao;
+
+    @Autowired
+    private UserDeveloperMapDAO userDeveloperMapDao;
+
+    @Autowired
+    private FF4j ff4j;
+
+    @SuppressWarnings("checkstyle:linelength")
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        if (!ff4j.check(FeatureList.ROLE_DEVELOPER)) {
+            LOGGER.fatal("The " + FeatureList.ROLE_DEVELOPER + " flag is OFF. Exiting job.");
+            return;
+        }
+
+        LOGGER.info("********* Starting the Invite Developers job *********");
+        setSecurityContext();
+        try {
+            List<CertificationStatusType> certificationStatuses = Stream.of(CertificationStatusType.Active,
+                    CertificationStatusType.SuspendedByAcb,
+                    CertificationStatusType.SuspendedByOnc).collect(Collectors.toList());
+            List<DeveloperDTO> allDevelopers = developerListingMapDao.getDevelopersWithListingsInEditionAndStatus(
+                    CertificationEditionConcept.CERTIFICATION_EDITION_2015.getId(), certificationStatuses);
+            LOGGER.info("There are " + allDevelopers.size() + " in the system with Active or Suspended 2015 listings.");
+            allDevelopers.sort(new Comparator<DeveloperDTO>() {
```

Comment:
What's the point of the sort? I imagine it doesn't hurt anything, but am curious

Reply:
As I was working on the job I was looking at the logs a lot and it was just MUCH NICER to read through them with the developers in an order that a human might reasonable expect. I expect QA will use them in DataDog and thought they would appreciate it as well so left it in.

Reply:
Seems reasonable